### PR TITLE
py-lightly: add v1.4.8

### PR DESCRIPTION
**Pull request by adamjstewart**
https://github.com/lightly-ai/lightly/releases/tag/v1.4.8
https://github.com/lightly-ai/lightly/releases/tag/v1.4.9

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-lightly/package.py [hunk 1]
```diff
@@ -37,6 +38,13 @@ class PyLightly(PythonPackage):
     depends_on("py-tqdm@4.44:", type=("build", "run"))
     depends_on("py-urllib3@1.15.1:", type=("build", "run"))
 
+    # requirements/openapi.txt
+    depends_on("py-python-dateutil@2.5.3:", when="@1.4.8:", type=("build", "run"))
+    depends_on("py-setuptools@21:", when="@1.4.8:", type=("build", "run"))
+    depends_on("py-urllib3@1.25.3:", when="@1.4.8:", type=("build", "run"))
+    depends_on("py-pydantic@1.10.5:1", when="@1.4.8:", type=("build", "run"))
```
Comment (manuelakuhn) [inline var/spack/repos/builtin/packages/py-lightly/package.py:45]: The newest version available in spack is `py-pydantic@1.10.2`

File: var/spack/repos/builtin/packages/py-lightly/package.py [hunk 2]
```diff
@@ -37,6 +38,13 @@ class PyLightly(PythonPackage):
...
+    depends_on("py-aenum@3.1.11:", when="@1.4.8:", type=("build", "run"))
```
Comment (manuelakuhn) [inline var/spack/repos/builtin/packages/py-lightly/package.py:46]: The newest version available in spack is `py-aenum@2.1.2`

**Review by manuelakuhn — COMMENTED** [review]
Other than the dependency versions that are not in spack yet this looks good to me.

File: var/spack/repos/builtin/packages/py-pydantic/package.py [hunk 3]
```diff
@@ -12,16 +12,16 @@ class PyPydantic(PythonPackage):
...
     pypi = "pydantic/pydantic-1.8.2.tar.gz"
 
+    version("1.10.9", sha256="95c70da2cd3b6ddf3b9645ecaa8d98f3d80c606624b6d245558d202cd23ea3be")
     version("1.10.2", sha256="91b8e218852ef6007c2b98cd861601c6a09f1aa32bbbb74fab5b1c33d4a1e410")
     version("1.9.2", sha256="8cb0bc509bfb71305d7a59d00163d5f9fc4530f0881ea32c74ff4f74c85f3d3d")
     version("1.8.2", sha256="26464e57ccaafe72b7ad156fdaa4e9b9ef051f69e175dbbb463283000c05ab7b")
 
     variant("dotenv", default=False, description="Install requirements for pydantic.dotenv")
 
-    depends_on("python@3.7:", when="@1.10.2:", type=("build", "run"))
```
Comment (manuelakuhn) [inline var/spack/repos/builtin/packages/py-pydantic/package.py:21]: Isn't this giving problems if we remove all of the python@3.7 restrictions but spack still supports python@3.6?

File: var/spack/repos/builtin/packages/py-pydantic/package.py [hunk 4]
```diff
@@ -12,16 +12,16 @@ class PyPydantic(PythonPackage):
...
-    depends_on("python@3.6.1:", type=("build", "run"))
     depends_on("py-setuptools", type="build")
-    depends_on("py-typing-extensions@4.1.0:", when="@1.10.2:", type=("build", "run"))
+    depends_on("py-typing-extensions@4.2:", when="@1.10.9:", type=("build", "run"))
+    depends_on("py-typing-extensions@4.1:", when="@1.10.2:", type=("build", "run"))
```
Comment (manuelakuhn) [inline var/spack/repos/builtin/packages/py-pydantic/package.py]:
```suggestion
    depends_on("py-typing-extensions@4.1:", when="@1.10:", type=("build", "run"))
```
just nitpicking here :-)  (is also fine as is)

**Review by manuelakuhn — DISMISSED** [review]

**Review by manuelakuhn — DISMISSED** [review]

**Comment by adamjstewart** [comment]
@spackbot run pipeline

**Comment by adamjstewart** [comment]
@spackbot run pipeline

**Review by manuelakuhn — APPROVED** [review]

**Pull request merged**